src/rulesets/oas/__tests__/typed-enum.ts

```diff
@@ -1,11 +1,13 @@
 import { DiagnosticSeverity } from '@stoplight/types';
-import { typedEnum } from '../../../functions/typedEnum';
+import { functions } from '../../../functions';
 import { RuleType, Spectral } from '../../../index';
+import { setFunctionContext } from '../../evaluators';
+import { typedEnum } from '../functions/typedEnum';
 import { rules } from '../index.json';
 
 describe('typed-enum', () => {
   const s = new Spectral();
-  s.setFunctions({ typedEnum });
+  s.setFunctions({ typedEnum: setFunctionContext({ functions }, typedEnum) });
```

Comment:
@P0lip re `setFunctionContext` , none of the other custom function test files leverage it. When should it be used?

Reply:
This is how we inject these built-in functions.
It's needed to use `setFunctionContext` if your custom functions makes use of the built-in one.
Spectral takes care of it [here](https://github.com/stoplightio/spectral/blob/develop/src/spectral.ts#L176), but obviously if you have a unit test, you need to do it on your own.